### ISSUE TITLE
Original Carlinkit (10MB rootfs) + latest firmware

### ISSUE THREAD

**Issue by bookemdano08** [ludwig-v/wireless-carplay-dongle-reverse-engineering#137]
I still have one of the early (circa 2019) Carlinkit dongles. It's been sitting in a drawer since I got a v2 over a year ago (which continues to work great in my Ford). I pulled out this old one recently to give to my sister, but wanted to try to get it current on firmware. It was running 2020.08.29. Using the OTA method I was able to get it to 2021.08.30 (involved flashing several intermediate versions).

It's now offering me 2021.12.22.1643, which I have not seen posted here, nor at the grabher-industries archive. Trying to flash it OTA resulted in an "Upgrade Failed" error after a few seconds. It's not bricked after that, just continues running on 2021.08.30. I've attempted the OTA upgrade a half dozen times (tried resetting the adapter first as well as using an iPad usb plug rather than out in the car) and it always fails.

Managed to obtain the 2021.12.22.1643 img file by downloading it from: http://112.74.112.110/mnt/downloads/2022-01-24/U2W_Update_2021.12.22.1643.img but unfortunately it won't flash via USB either. Just reboots the adapter after 15 seconds or so.

That firmware is less than 500KB in size (sort of like the 2021.07.04 one, although that flashed fine for me), so I don't know if it's just corrupted or is not compatible with my old 10MB rootfs dongle.

For grins, I tried to flash the 2022.01.24 img I found on grabher-industries. It flashed but the process only took about 15 seconds, which is odd since that firmware is 11MB in size. To my surprise, the adapter "works" in that CarPlay comes up in the car, but the 192.168.50.2 website is borked. It's got the new interface but displays almost no data. None of the settings can be changed and the Update button just brings up a "Loading" modal (in Chinese) that never goes away. The only thing that actually seems to work on the web page is the Reset button, which does indeed reset the dongle, but doesn't solve the issues with the web page.

So I guess I'm wondering if ludwig-v or anyone else here has had a similar experience with the old-style adapters. Of course I can leave it on 2021.08.30 since I think that's a pretty stable firmware, but I would like to be able to get the new web UI since it has a bunch more information on it.

**tl;dr: Anyone successful in flashing anything newer than 2021.08.30 on an old-style adapter (green UI + attached USB A cable)? If so, how did you do it and is it fully working? 2022.01.24 flashes onto mine (via USB) but 192.168.50.2 is borked.**

Maybe the 10MB rootfs is the problem? I noticed that 2022.01.24 is the first non-Autokit firmware that is > 10MB.

**Comment by simon0117** [comment]
I just reported issue #144 here and then found this which I now see is the same problem I had with the failed upgrade, just different car and dongle brand. "borked" website and Chinese "loading" modal that won't go away. I can help troubleshoot if I knew how :-)

**Comment by ste-sanjaysingh** [comment]
+ Following : Experiencing same issue with Toyota and Cplay2air 2018 dongle.

**Comment by ludwig-v** [comment]
Your best bet with old units is to flash a full dump directly because you will have nothing but problems if your rootFS is still at 10MB.
https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/blob/master/Flash_Dump/
